### DISCUSSION TITLE
androidkindlekey.py can't get key

### DISCUSSION THREAD

**Discussion by RobertWielewicki** [apprenticeharper/DeDRM_tools#1743] [Q&A]
I have downloaded newest DeDRM and unpacked it to get to androidkindlekey.py.
I have connected phone to PC, enabled USB debugging. Ebooks of interest are already downloaded and copied to home directory.
I did "adb backup com.amazon.kindle"
backup.ab appeared
I tried it with sudo and without, python2 (which doesnt work) and python3 (which outputs error, but at least launches)

python androidkindlekey.py -b /home/robert/backup.ab /home/robert/key
androidkindlekey.py v2.0
Copyright © 2010-2020 Thom, Apprentice Harper et al.

Could not retrieve Kindle for Android key.

---------------------
Is there a way to get the key required to get access to these ebooks from Android? I dont want to setup VM just to install windows, just to use old Kindle for PC to do it.

**Comment by j-howell** [comment]
From the [FAQ](https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#why-dont-the-tools-work-with-kindle-for-android-ebooks):

**Why don't the tools work with Kindle for Android ebooks?**

Amazon turned off backup for Kindle for Android, so the tools can no longer find the key. The workaround is to install Kindle for PC or Kindle for Mac and use books from there instead.

**Reply by RobertWielewicki** [reply]
Thank you for response. I guess this is what I will do, unfortunally.
If somebody knows of workaround (prefferably without root, but I can root my phone), please let me know.

**Reply by Leseratte10** [reply]
Dumping app data from the newest Kindle for Android is possible on an unrooted phone, I've done that a couple weeks ago. Though the process is not exactly easy - you need to decompress the Kindle apk, then change the "allowBackup" switch in the manifest from false to true, then re-pack, re-sign and re-install that APK. I don't think there's complete instructions on how to do that anywhere, if you're still interested I could try to write them up somewhere. 
So far I only tested that on Linux, but looking at your earlier comment I assume you're already running Linux anyways. 
What I have not tested is the file format of books dumped from the newest Kindle for Android - if these are KFX files then the key probably won't help you. 
I also have not tested if the app backup for the newest K4A still contains the keys in the same format as before or if they changed that around, too.
